substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -1964,7 +1970,7 @@ private List<DebugLocalValueInfo> initSyntheticInfoList(ParamLocationProducer lo
                 JavaKind kind = ownerType.getJavaKind();
                 JavaKind storageKind = isForeignWordType(ownerType, ownerType) ? JavaKind.Long : kind;
                 assert kind == JavaKind.Object : "must be an object";
-                NativeImageDebugLocalValue value = locProducer.nextLocation(kind);
+                NativeImageDebugLocalValue value = locProducer.thisLocation();
                 debugContext.log(DebugContext.DETAILED_LEVEL, "locals[%d] %s type %s slot %d", localIdx, name, ownerType.getName(), slot);
                 debugContext.log(DebugContext.DETAILED_LEVEL, "  =>  %s kind %s", value, storageKind);
                 localInfos.add(new NativeImageDebugLocalValueInfo(name, value, storageKind, ownerType, slot, firstLine));
```

Comment:
There is a small bug in the next line that I found while testing the PR. I've noticed that some arguments in synthetic frames are reported as optimized out, and the reason is that we're miscounting slots for `Word` type arguments: in the next line we use `storageKind` to count the slots, but it should be `kind`.
```diff
-                slot += storageKind.getSlotCount();
+                slot += kind.getSlotCount();
```
This also reminded me that we should test this. So could you please add tests that check that we can list the arguments of (1) an entry point (they use a different calling convention than Java now), e.g., I think checking `main` (the entry point for `JavaMainWrapper.run`) should be sufficient, and (2) a method with a mixture of `Word` and non-`Word` type arguments (as a regression test for this bug).

Reply:
Done

Reply:
I still need to update the tests.

Reply:
Yes, something like that, but for a method that has more arguments after the first `WordBase` type argument (e.g., like `CEntryPointSnippets.createIsolate`), because if you were to set a breakpoint on such a method, you wouldn't be able to see those arguments without this fix.

Reply:
I just pushed a change to test the mixed argument case. It recognizes all arguments with the patch to `initSyntheticInfoList` in place (i.e. incrementing slot count using `kind.getSlotCount()`). It fails I revert that specific fix (i.e. increment by `storageKind.getSlotCount()`). So, this looks like a wrap.

Reply:
Yes, looks good, thanks!